pkg/metrics/custom_metrics.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2025 The Aibrix Team.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var (
+	customGauges     = make(map[string]*prometheus.GaugeVec)
+	customGaugesMu   sync.RWMutex
+	customCounters   = make(map[string]*prometheus.CounterVec)
+	customCountersMu sync.RWMutex
+
+	// Function variable that can be overridden for testing
+	SetGaugeMetricFnForTest = defaultSetGaugeMetric
```

Comment:
Seems there's no similar one for counter? Do we need something like `SetCounterMetricFnForTest`?

Reply:
It can be added later when the test have some counter coverage?

Reply:
good catch, added tests for counters as well.

Reply:
looks good to me